Description: providing ant targets to build on ppc64 and ppc64el
Author: Sylvestre Ledru <sylvestre@debian.org>
Forwarded: sgothel@jausoft.com
Last-Update: 2023-04-17

--- a/make/build-jogl.xml
+++ b/make/build-jogl.xml
@@ -1481,6 +1481,12 @@
       <property name="linker.cfg.id.os"                    value="linker.cfg.linux.jogl.x11" />
     </target>
 
+    <target name="c.configure.linux.ppc64" if="isLinuxPpc64">
+      <echo message="Linux.PPC64" />
+      <property name="compiler.cfg.id"                     value="compiler.cfg.linux" />
+      <property name="linker.cfg.id.os"                    value="linker.cfg.linux.jogl.x11" />
+    </target>
+
     <target name="c.configure.linux.s390" if="isLinuxs390">
       <echo message="Linux.S390" />
       <property name="compiler.cfg.id"                     value="compiler.cfg.linux" />
@@ -1499,7 +1505,7 @@
       <property name="linker.cfg.id.os"                    value="linker.cfg.linux.jogl.x11" />
     </target>
 
-    <target name="c.configure.linux" depends="c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.ia64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.ppc,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.x11" if="isLinux" />
+    <target name="c.configure.linux" depends="c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.ia64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.ppc,c.configure.linux.ppc64,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.x11" if="isLinux" />
 
     <target name="c.configure.android" if="isAndroid">
       <echo message="Android" />
--- a/make/build-nativewindow.xml
+++ b/make/build-nativewindow.xml
@@ -760,6 +760,12 @@
       <property name="linker.cfg.id.oswin"                  value="linker.cfg.linux.nativewindow.x11" />
     </target>
 
+    <target name="c.configure.linux.ppc64" if="isLinuxPpc64">
+      <echo message="Linux.PPC64" />
+      <property name="compiler.cfg.id"                      value="compiler.cfg.linux" />
+      <property name="linker.cfg.id.oswin"                  value="linker.cfg.linux.nativewindow.x11" />
+    </target>
+
     <target name="c.configure.linux.s390" if="isLinuxs390">
       <echo message="Linux.S390" />
       <property name="compiler.cfg.id"                      value="compiler.cfg.linux" />
@@ -781,7 +787,7 @@
       <property name="linker.cfg.id.drm"                    value="linker.cfg.linux.nativewindow.drm" />
     </target>
 
-    <target name="c.configure.linux" depends="c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.ia64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.ppc,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.x11" if="isLinux" />
+    <target name="c.configure.linux" depends="c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.ia64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.ppc,c.configure.linux.ppc64,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.x11" if="isLinux" />
 
     <target name="c.configure.android" if="isAndroid">
       <echo message="Android" />
--- a/make/build-newt.xml
+++ b/make/build-newt.xml
@@ -826,6 +826,16 @@
       <echo message="linker.cfg.id.oswin ${linker.cfg.id.oswin}" />
     </target>
 
+    <target name="c.configure.linux.ppc64" if="isLinuxPpc64">
+      <echo message="Linux.ppc64" />
+      <property name="compiler.cfg.id"                      value="compiler.cfg.linux" />
+      <condition property="linker.cfg.id.oswin"             value="linker.cfg.linux.newt.x11"
+                                                            else="linker.cfg.linux">
+          <isset property="isX11" />
+      </condition>
+      <echo message="linker.cfg.id.oswin ${linker.cfg.id.oswin}" />
+    </target>
+
     <target name="c.configure.linux.s390" if="isLinuxs390">
       <echo message="Linux.S390" />
       <property name="compiler.cfg.id"                      value="compiler.cfg.linux" />
@@ -863,7 +873,7 @@
       <property name="linker.cfg.id.drm"                    value="linker.cfg.linux.newt.drm"/>
     </target>
 
-    <target name="c.configure.linux" depends="c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.ppc,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.linux.ia64,c.configure.x11" if="isLinux"/>
+    <target name="c.configure.linux" depends="c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.ppc,c.configure.linux.ppc64,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.linux.ia64,c.configure.x11" if="isLinux"/>
 
     <target name="c.configure.android" if="isAndroid">
       <echo message="Android" />
